File: core/src/com/exp4j/Functions/RandomFunction.java
package com.exp4j.Functions;

import java.util.Random;

import net.objecthunter.exp4j.function.Function;

import com.badlogic.gdx.math.MathUtils;

public class RandomFunction extends Function
{
	private Random ran;

	public RandomFunction()
	{
		super( "rnd", 1 );
	}

	public RandomFunction( Random ran )
	{
		super( "rnd", 1 );

		this.ran = ran;
	}

	@Override
	public double apply( double... arg0 )
	{
		if ( ran != null )
		{
			return ran.nextFloat() * arg0[0];
		}
		else
		{
			return MathUtils.random() * arg0[0];
		}
	}

}

File: core/src/com/exp4j/Functions/RandomFunction.kt
package com.exp4j.Functions

import com.badlogic.gdx.utils.Pool
import net.objecthunter.exp4j.function.Function
import squidpony.squidmath.LightRNG

class RandomFunction : Function("rnd", 1)
{
	private val ran = LightRNG()

	fun set(seed: Long)
	{
		this.ran.setSeed(seed)
	}

	override fun apply(vararg arg0: Double): Double
	{
		return ran.nextFloat() * arg0[0]
	}

	fun free()
	{
		synchronized(pool)
		{
			pool.free(this)
		}
	}

	companion object
	{

		val pool = object : Pool<RandomFunction>() {
			override fun newObject(): RandomFunction
			{
				return RandomFunction()
			}
		}

		fun obtain(): RandomFunction
		{
			synchronized(pool)
			{
				return pool.obtain()
			}
		}
	}
}
